deploy/02_Deploy_APIConsumer.js

```diff
@@ -23,18 +25,23 @@ module.exports = async ({
   }
   const jobId = networkConfig[chainId]['jobId']
   const fee = networkConfig[chainId]['fee']
+  const networkName = networkConfig[chainId]['name']
 
   const apiConsumer = await deploy('APIConsumer', {
     from: deployer,
     args: [oracle, jobId, fee, linkTokenAddress],
     log: true
   })
 
-  log("Run the following command to fund contract with LINK:")
-  log("npx hardhat fund-link --contract " + apiConsumer.address + " --network " + networkConfig[chainId]['name'] + additionalMessage)
-  log("Then run API Consumer contract with following command:")
-  log("npx hardhat request-data --contract " + apiConsumer.address + " --network " + networkConfig[chainId]['name'])
+  //try to auto fund contract with LINK
+  const autoFund = await autoFundCheck(apiConsumer.address, networkName, linkTokenAddress, additionalMessage);
```

Comment:
This whole autoFund thing confuses me. Why not just fund it if `apiConsumer.newlyDeployed`?

Reply:
what if you're deploying to mainnet? Or what if there's no LINK in the wallet? The auto fund is doing what you just suggested, except its configurable so it can be turned on/off per network, and it has some smarts to check if there's LINK in the wallet as well